### PR TITLE
V3 updates

### PR DESCRIPTION
**Pull request by kreed0422**
Harry,  can you be my second set of eyes and review 'fix: logic for accepting past/future dates, set errors, unit tests' this commit.

Let me know if I can merge and build another release of the library.

Thanks

### CODE REVIEW

File: projects/common/lib/components/date/date.component.ts [hunk 1]
```diff
@@ -255,9 +264,12 @@ You must use either [restrictDate] or the [dateRange*] inputs.
 
       // Date function appears to use setYear() so any year 0-99 results in year 1900 to 1999
       // Set each field individually, use setFullYear() instead of setYear()
-      this.date = new Date();
+      // Set time on date to 00:00:00 for comparing later
+      this.date = startOfDay(new Date());
       this.date.setFullYear(year);
       this.date.setMonth(month);
+      this.date.setFullYear(year); // To correct year when value has less than 4 characters
```
Comment (harrymaynard-maximus) [inline projects/common/lib/components/date/date.component.ts]: This is a duplicate of line 269.

File: projects/common/lib/components/date/date.component.ts [hunk 2]
```diff
@@ -103,7 +115,7 @@ export class DateComponent extends AbstractFormControl
   _defaultErrMsg: ErrorMessage = {
     required: `${LabelReplacementTag} is required.`,
     dayOutOfRange: `Invalid ${LabelReplacementTag}.`,
-    yearDistantPast: `Invalid ${LabelReplacementTag}.`,
+    yearDistantPast: `Invalid ${LabelReplacementTag}`,
```
Comment (harrymaynard-maximus) [inline projects/common/lib/components/date/date.component.ts]: Is this an intentional removal. All the remaining error messages end with a period.

File: projects/common/lib/components/date/date.component.spec.ts [hunk 3]
```diff
@@ -67,7 +88,8 @@ describe('DateComponent', () => {
       expect( getLegendContext( fixture, 'common-date', 'date1') ).toBe( component.defaultLabel );
     }));
 
-    it('should remove invalidValue error when all fields are reset.', fakeAsync(() => {
+    // TODO: Revisit - not quite correct
+    xit('should remove invalidValue error when all fields are reset.', fakeAsync(() => {
```
Comment (harrymaynard-maximus) [inline projects/common/lib/components/date/date.component.spec.ts]: This should be included (is it passing?).

Comment (kreed0422) [reply projects/common/lib/components/date/date.component.spec.ts]: Needs to be updated - figured out how to trigger inputs using dispatchEvent()

**Pull request merged**